src/main/kotlin/no/nb/mlt/wls/product/repository/ProductRepository.kt

```diff
@@ -6,4 +6,6 @@ import org.springframework.data.mongodb.repository.MongoRepository
 
 interface ProductRepository : MongoRepository<Product, String> {
     fun findByHostName(hostName: HostName): List<Product>
+
+    fun existsByHostId(id: String): Boolean
```

Comment:
Now that I think about it, I think we need to check existence both by host ID and name, since we can have duplicate IDs across different hosts